traveler-ios-ui/TravelerKitUI/Models/ParkingResultContext.swift

```diff
@@ -14,38 +14,45 @@ public protocol ParkingResultContextObserving: class {
     func parkingResultContextDidChangeSelectedIndex(_ context: ParkingResultContext)
 }
 
-public class ParkingResultContext {
+public class ParkingResultContext: ObservingContext {
     private(set) var spots: [ParkingSpot]?
 
     var selectedIndex: Int? {
         didSet {
-            observers.forEach {
-                $0.parkingResultContextDidChangeSelectedIndex(self)
-            }
+            notifyObserversWithSelectedIndexChange()
```

Comment:
You're doing notifyObserversWithResultUpdate on a background thread but not here, why?.